Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2987,11 +2987,23 @@ extension BrowserViewController: NewTabPageDelegate {
     guard let url = favorite.url else { return }
     switch action {
     case .opened(let inNewTab, let switchingToPrivateMode):
-      navigateToInput(
-        url,
-        inNewTab: inNewTab,
-        switchingToPrivateMode: switchingToPrivateMode
-      )
+      if !privateBrowsingManager.isPrivateBrowsing, Preferences.Privacy.privateBrowsingLock.value {
+        self.askForLocalAuthentication { [weak self] success, error in
+          if success {
+            self?.navigateToInput(
+              url,
+              inNewTab: inNewTab,
+              switchingToPrivateMode: switchingToPrivateMode
+            )
+          }
+        }
+      } else {
```

Comment:
I'm confused by this logic (and the rest below): Why are we asking for auth without checking `switchingToPrivateMode` before?

Reply:
Yes because of the result of Auth we will let them switch or not. This is the requested logic, it is like this for tab tray also.

Reply:
Right but wouldn't this display an auth request to a user who is simply opening a new tab in general since you don't verify that the new tab is opening in a private tab first?

Reply:
No because this logic is only added to the actions opening private tab when user is in priva mode.

Reply:
I tested every action one by one hopefully didn't miss anything here but I will warn QA about to test all the actions mentioned in PR not just ticket.

Reply:
I just tested this and this logic is bugged:

1. Turn on Private Browsing Lock
2. Long press a favourite and choose "Open in New Tab"
3. See auth lock

Because you don't check `switchingToPrivateMode` along with whether or not private lock is on along with if you're already private browsing and if the pref is on

Reply:
Ok got it, I will check all the actions here again with both of the modes and correct mistakes.

Reply:
Fixed at ###  Bottom bar long press middle icon action open in new private tab
Thanks for checking in detail for this